pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -3625,6 +3626,15 @@ public Position getLastPosition() {
         return ledger.getLastConfirmedEntry();
     }
 
+    @Override
+    public CompletableFuture<Position> getLastDispatchablePosition() {
+        return ledger.asyncReverseFindPositionOneByOne((entry -> {
+            MessageMetadata md = Commands.parseMessageMetadata(entry.getDataBuffer());
+            // If a messages has marker will filter by AbstractBaseDispatcher.filterEntriesForConsumer
+            return !Markers.isServerOnlyMarker(md);
+        }));
+    }
```

Comment:
It is better to return a smaller value with `getMaxReadPosition()`.

Reply:
Why, since IO may be involved here, it is more appropriate to return to the future

Reply:
Oh, sorry, I mean if the result of `asyncReverseFindPositionOneByOne` is greater than `getMaxReadPosition`, then the result of `getMaxReadPosition` should be returned.

Reply:
> Oh, sorry, I mean if the result of asyncReverseFindPositionOneByOne is greater than getMaxReadPosition, then the result of getMaxReadPosition should be returned.

I  think the result of `asyncReverseFindPositionOneByOne` is never greater than `getMaxReadPosition()`